### ISSUE TITLE
short all-in raises can reopen betting

### ISSUE THREAD

**Issue by ctm** [ctm/mb2-doc#150]
When I fixed #143, I realized that mb2 doesn't currently record enough information to properly handle short all-in raises. Specifically, although a short all-in shouldn't re-open betting, it should be possible for one or more calls to reopen betting. So, my fix for #143 broke the code that was preventing short all-ins from reopening betting.

The fix is relatively straightforward, but it's more than a couple of lines of code and I need to do it some morning when I'm fresh.

**Comment by ctm** [comment]
Well, the nice thing is my code has this message:
```
            // TODO: We need the second condition (i.e., the one after
            // &&) to deal with a short-raise that hasn't re-opened
            // betting.  However, this probably opens up the ability
            // of *others* to raise when they should't be allowed to.
            // As such, when we detect a short-raise, I think we need
            // to write down some extra state so that what happens
            // afterward can be appropriately limited.
```
OTOH, @1pm Mountain, I'm not "fresh". OTOOH, I did have three double espressos earlier today. Whee.

**Comment by ctm** [comment]
Upon reflection, that comment is most likely wrong. I think the badness of the code I fixed for #159 explains all the short all-in bugs I've seen in the real world. I deleted the above comment and am closing this ticket, albeit with a small amount of trepidation. 

FWIW, #127 will be a large enough refactor that if there is a hidden bug, it might very well be squashed by the refactor. Additionally, I'll try to structure the new code so that I can add automated tests that actually use the values from the TDA's examples.
